File: library/src/main/scala/com/dmanchester/playfop/sapi/PlayFopComponents.scala
package com.dmanchester.playfop.sapi

import com.dmanchester.playfop.sinternal.PlayFopImpl

/** A trait for dependency-injecting PlayFOP into Scala applications at
  * compile time.
  */
trait PlayFopComponents {
  lazy val playFop: PlayFop = new PlayFopImpl()
}

File: library/src/main/java/com/dmanchester/playfop/japi/PlayFopComponents.java
package com.dmanchester.playfop.japi;

import com.dmanchester.playfop.jinternal.PlayFopImpl;

/**
 * An interface for dependency-injecting PlayFOP into Java applications at
 * compile time.
 */
public interface PlayFopComponents {

    default PlayFop playFop() {
        return new PlayFopImpl();
    }
}
